src/kmeans.jl

```diff
@@ -295,13 +299,15 @@ function update_centers!(
         if wj > 0
             cj = assignments[j]
             1 <= cj <= k || error("assignment out of boundary.")
-
             if to_update[cj]
-                xj = view(X, :, j)
                 if cweights[cj] > 0
-                    centers[:, cj] .+= xj * wj
```

Comment:
I guess this `*` (plus the one below) was the reason for regression in weighted case as it creates the new vector.
`.+= xj .* wj` should avoid that.
I'm a little bit hesitant about converting everything to explicit loops as it defeats the purpose of having broadcast support in the language.

Reply:
I'm happy with opening another PR for this kind of stuff after this one gets merged so that we can isolate changes related to the initial problem from other changes?

Reply:
(FWIW, I tried exactly what you said and it increases the allocation by a lot and doubles the time, possibly because it constructs `centers[:, cj]`. 
There may be better things to do with `copyto!` etc but I must admit I'd rather get this PR done with first...

Reply:
I thought `centers[:, cj] .+=` calls `setindex!()` directly, but it actually creates `centers[:, cj]` vector behind the scenes (`centers[:, cj] .=` doesn't). So the better version is `cej = view(centers, :, cj); cej .+= xj .* wj`.